oap-ape/ape-java/ape-server/src/main/java/com/intel/oap/ape/service/netty/server/NettyServer.java

```diff
@@ -0,0 +1,132 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Netty server to handle requests from clients.
+ * This server will run standalone.
+ */
+public class NettyServer {
+    private static final Logger LOG = LoggerFactory.getLogger(NettyServer.class);
+
+    private int port;
+    private final String address;
+
+    private ServerBootstrap bootstrap;
+    private ChannelFuture bindFuture;
+
+    public NettyServer(int port, String address) {
+        this.port = port;
+        this.address = address;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void run() throws Exception {
+        final long start = System.nanoTime();
+
+        EventLoopGroup bossGroup = new NioEventLoopGroup();
+        EventLoopGroup workerGroup = new NioEventLoopGroup();
+        ChannelHandler encoder = new NettyMessage.NettyMessageEncoder();
+
+        bootstrap = new ServerBootstrap();
+        bootstrap.group(bossGroup, workerGroup)
+                .channel(NioServerSocketChannel.class)
+                .childHandler(new ChannelInitializer<SocketChannel>() {
+                    @Override
+                    public void initChannel(SocketChannel ch) throws Exception {
+                        ch.pipeline().addLast(
+                                encoder,
+                                new NettyMessage.NettyMessageDecoder(),
+                                new RequestHandler());
+                    }
+                })
+                .option(ChannelOption.SO_BACKLOG, 128)
+                .childOption(ChannelOption.SO_KEEPALIVE, true);
+
+        // Bind and start to accept incoming connections.
+        bootstrap.localAddress(InetAddress.getByName(address), port);
+        bindFuture = bootstrap.bind().sync();
+
+        InetSocketAddress localAddress = (InetSocketAddress) bindFuture.channel().localAddress();
+        port = localAddress.getPort();
+        final long duration = (System.nanoTime() - start) / 1_000_000;
+        LOG.info("Successful initialization (took {} ms). Listening on SocketAddress {}.",
+                duration, localAddress);
+    }
+
+    public void shutdown() {
+        final long start = System.nanoTime();
+        if (bindFuture != null) {
+            bindFuture.channel().close().awaitUninterruptibly();
+            bindFuture = null;
+        }
+
+        if (bootstrap != null) {
+            bootstrap.config().group().shutdownGracefully();
+            bootstrap.config().childGroup().shutdownGracefully();
+            bootstrap = null;
+        }
+        final long duration = (System.nanoTime() - start) / 1_000_000;
+        LOG.info("Successful shutdown (took {} ms).", duration);
+    }
+
+    private static String getDefaultBindAddress() {
+        return new InetSocketAddress(0).getAddress().getHostAddress();
+    }
+
+    public static void main(String[] args) throws Exception {

```

Comment:
could remove main method? or add some comments to indicate just for debug/test.

Reply:
Not for debug/test. This is the `main` process of the server.